CHANGES.rst

```diff
@@ -1,6 +1,266 @@
+  unless ``allow_none=True`` is specified. (#1432)
+* The ``Date`` trait type has a new argument ``allow_datetime``. In the future,
+  ``datetime`` instances will not be valid values for a ``Date`` trait unless
+  ``allow_datetime=True`` is specified. (#1429)
+
+
+Performance
+~~~~~~~~~~~
+
+* The ``ObserverGraph`` instances that result from compiling
+  ``ObserverExpression`` objects and observe mini-language strings are now
+  cached. This should speed up creation and instantiation of ``HasTraits``
+  subclasses that involve listening for the same pattern in multiple places.
+  (#1516, #1528)
+* The equality definition on ``ObserverExpression`` has been simplified.
+  (#1517)
+* The ``ObserverExpression``, ``ObserverGraph`` and related
+  classes now use ``__slots__`` to improve speed and memory use. (#1513, #1515)
+* The ``on_trait_change`` method has been sped up by almost a factor of two,
+  by removing unnecessary internal usage of Traits in the parsing and listener
+  functionality. (#1490, #1491, #1492, #1493)
+
+
+Changes
+~~~~~~~
+
+* An invalid static default value in a ``PrefixList`` or ``PrefixMap`` trait
+  declaration now raises ``ValueError`` rather than ``TraitError``. (#1564)
+* ``PrefixList`` and ``PrefixMap`` no longer cache completions. (#1564)
+* A failure to parse an ``observe`` mini-language string now raises
+  ``ValueError`` rather than ``LarkError``. (#1507)
+* The ``NotifierNotFound`` exception is now published in
+  ``traits.observation.api``. (#1498)
+* An attempt to access a nonexistent "dunder" attribute (an attribute whose
+  name starts and ends with "__") on a ``CTrait`` instance will now raise
+  ``AttributeError``. Previously, it would return ``None``. (#1469, #1474,
+  #1477)
+
+
+Deprecations
+~~~~~~~~~~~~
+
+* The ``Any`` trait type currently implicitly makes a per-``HasTraits``-instance
+  copy of the default value if that value is an instance of either ``list`` or
+  ``dict``. This behaviour is deprecated, and will be removed in Traits 7.0.
+  For a per-instance default, use the new ``factory`` argument to ``Any``
+  instead. (#1548, #1532)
+* The ``Date``, ``Datetime`` and ``Time`` trait types will no longer accept
+  ``None`` as a valid trait value in the future. To keep the existing
+  behaviour, use the new ``allow_none`` keyword argument to these trait types.
+  (#1444)
+* The ``Date`` trait type will no longer accept ``datetime`` instances by
+  default in the future. To keep the existing behaviour, use the new
+  ``allow_datetime`` keyword argument. (#1441)
+* The ``Symbol`` trait type is deprecated. For resolution of a string
+  representing a package/module/object combination, use ``import_symbol``
+  instead. (#1542)
+* The ``MetaHasTraits.add_listener`` and ``MetaHasTraits.remove_listener``
+  methods are deprecated. If you need the functionality that these methods
+  provide, please discuss your use-case with the ETS maintenance team. (#1550)
```

Comment:
```suggestion
  provide, please discuss your use-case with the ETS maintenance team. (#1550)
```

should we be asking people to open an issue? i'm not sure what "discuss" means here for external users.